### ISSUE TITLE
safeDump automatically reformat yaml strings

### ISSUE THREAD

**Issue by guanw** [nodeca/js-yaml#443]
I have question regarding the behavior of js-yaml with safeDump, we have a yaml editor which use the following method to generate yaml from js object:
```
self.generateYAML = function (graph) {
                if (graph === undefined) {
                    return '';
                }
                try {
                    return jsyaml.safeDump(graph, {
                        noCompatMode: true // fix auto quoting 'y'.
                    }).trim();
                } catch (e) {
                    var error = 'YAML validation error: ' + e.message;
                    console.log(error);
                    return '';
                }
            };
```

graph is passed with an object that has the following content:

```
'Fn::Base64':

  'Fn::Join':

    - "\n"

    -

      - scriptFile=~/createSchema.sh

      - scriptLog=~/createSchema.log

      - rm -f ${scriptFile}

      - touch ${scriptFile}

      - touch ${scriptLog}

      - echo "sqlplus / as sysdba << EOF" >> ${scriptFile}

      - echo "drop user c##myappuser cascade;" >> ${scriptFile}

      - echo "create user c##myappuser identified by ${schemaPassword};" >> ${scriptFile}

      - echo "grant create session, create table, unlimited tablespace to c##myappuser;" >> ${scriptFile}

      - echo "quit" >> ${scriptFile}

      - echo "EOF" >> ${scriptFile}

      - sh ${scriptFile} >> ${scriptLog}

```

Somehow after safedump, the line break was reformated (though it's still correct) and yaml becomes the following
![image](https://user-images.githubusercontent.com/7311035/45368994-deb4d800-b5b2-11e8-963b-584b0ecfda56.png)

The version we used is 3.10.0. Please let me know if there's anything I can do to disable this automatic transformation. Thanks.

**Comment by puzrin** [comment]
https://github.com/nodeca/js-yaml#safedump-object---options-

set `lineWidth` to `Infinity`

**Comment by guanw** [comment]
Hi, @puzrin Tried setting lineWidth but doesn't work. I also notice that if replace `"\n"` to `'\n'` there's no automatic conversion happen. Do you have any idea why and if it's a config issue or bug for ace editor?

**Comment by puzrin** [comment]
Try to create a repo with minimal possible code to reproduce your problem.

**Comment by guanw** [comment]
Just unzip the file and open index.html and check console. The result is reformatted after dump.
[js-yaml.zip](https://github.com/nodeca/js-yaml/files/2389684/js-yaml.zip)

**Comment by guanw** [comment]
I just realized it's not an issue with safeDump but probably the way safeLoad works.
For template that contains `"\n"`, it will be converted to enter key, but for `'\n'` it will be treated as `"\\n"`, and I just took a look here http://docs.octoprint.org/en/master/configuration/yaml.html. Seems like the behavior is correct on js-yaml. Thanks for helping. The template user provides is invalid in our case here.
